.manage.py

```diff
@@ -0,0 +1,47 @@
+from argparse import ArgumentParser
+from contextlib import chdir
+from pathlib import Path
+from shutil import rmtree, move
+from subprocess import call
+from tempfile import TemporaryDirectory
+
+VERSION = '3.12'
+
+
+def update_pots() -> None:
+    """updates translation sources"""
+    _call('git diff --exit-code')
+    with TemporaryDirectory() as directory:
+        with chdir(directory):
+            _clone_cpython_repo(VERSION)
+            _build_gettext()
+        rmtree('.pot')
+        move(Path(directory) / 'cpython/Doc/locales/pot', '.pot')
+    if (call("git diff -I'^\"POT-Creation-Date: ' --exit-code", shell=True)) != 0:
+        _call('git add .')
+        _call('git commit -m "Update sources"')
```

Comment:
If you are willing to git-add everything when at least on PO files changed is not just "POT-Creation-Date" date update, ok.

However if you do not want to add files the just have "POT-Creation-Date" date update, then `git add .` must be more elaborated. See how I did in python-tx-translations workflow:

```shell
# Enable globstar as it seems it is required for '**' to work (looks like to come disabled in GitHub Actions)
shopt -s globstar

# git add all changed files but the ones of only "POT-Creation-Date" date change
git diff -I'^"POT-Creation-Date: ' --numstat **/*.po **/*.pot | cut -f3 | xargs -r git add

# Add untracked files the may have been newly created.
git add $(git ls-files -o --exclude-standard *.po *.pot) .tx/config

# If something was added, then git commit. Otherwise, silently ignores with exit status 0
git diff-index --cached --quiet HEAD || { git commit -m "Update translations from ${{ env.BRANCH }}" && git push; }
```

You obviously don't need to store pot files as I did in there.

Reply:
Thank you for sharing the script! I pythonized it and added on the branch.